tests/raster/test_netcdf.py

```diff
@@ -0,0 +1,548 @@
+
+    def test_time_units(self):
+        """
+        Test time series and time units.
+        """
+        data = np.ones((100, 100, 100), dtype=np.float32)
+        dims = ['time', 'y', 'x']
+        coords = {'time': pd.date_range('2000-01-01', periods=data.shape[0])}
+
+        ds_ref = xr.Dataset({'sig': (dims, data),
+                             'inc': (dims, data)}, coords=coords)
+
+        time_units = 'days since 2000-01-01 00:00:00'
+        with NetCdfXrFile(self.filepath, mode='w', attrs={'time': {'units': time_units}}) as nc:
+            nc.write(ds_ref)
+
+        with NetCdfXrFile(self.filepath) as nc:
+            ds = nc.read()
+            np.testing.assert_array_equal(pd.DatetimeIndex(ds['time'].data),
+                                          coords['time'])
+
+    def test_geotransform(self):
+        """
+        Test computation of x and y coordinates.
+        """
+        xdim = 100
+        ydim = 200
+        data = np.ones((100, ydim, xdim), dtype=np.float32)
+        dims = ['time', 'y', 'x']
+        coords = {'time': pd.date_range('2000-01-01', periods=data.shape[0])}
+
+        ref_ds = xr.Dataset({'sig': (dims, data),
+                              'inc': (dims, data)}, coords=coords)
+
+        geotrans = (3000000.0, 500.0, 0.0, 1800000.0, 0.0, -500.0)
+        with NetCdfXrFile(self.filepath, mode='w', geotrans=geotrans) as nc:
+            nc.write(ref_ds)
+
+        with NetCdfXrFile(self.filepath) as nc:
+            ds = nc.read()
+
+        x = geotrans[0] + (0.5 + np.arange(xdim)) * geotrans[1] + \
+            (0.5 + np.arange(xdim)) * geotrans[2]
+        y = geotrans[3] + (0.5 + np.arange(ydim)) * geotrans[4] + \
+            (0.5 + np.arange(ydim)) * geotrans[5]
+
+        np.testing.assert_array_equal(ds['x'].values, x)
+        np.testing.assert_array_equal(ds['y'].values, y)
+
+    def test_non_temporal_read_and_write(self):
+        """ Test read and write for a dataset not containing any temporal information. """
+        data = np.ones((100, 100, 100), dtype=np.float32)
+        dims = ['layer', 'y', 'x']
+        coords = {'layer': range(data.shape[0])}
+        attr1 = {'unit': 'dB'}
+        attr2 = {'unit': 'degree'}
+
+        ds_ref = xr.Dataset({'sig': (dims, data, attr1),
+                             'inc': (dims, data, attr2),
+                             'azi': (dims, data, attr2)}, coords=coords)
+
+        with NetCdfXrFile(self.filepath, mode='w', data_variables=['sig', 'inc', 'azi'], stack_dims={'layer': None},
+                         nodatavals={'inc': -9999, 'azi': -9999}, dtypes={'inc': 'int32', 'azi': 'int32'}) as nc:
+            nc.write(ds_ref)
+
+        with NetCdfXrFile(self.filepath) as nc:
+            ds = nc.read()
+            np.testing.assert_array_equal(ds['sig'][:], ds_ref['sig'][:])
+            np.testing.assert_array_equal(ds['inc'][:], ds_ref['inc'][:])
+            np.testing.assert_array_equal(ds['azi'][:], ds_ref['azi'][:])
```

Comment:
Same as for `NetCdf4Test` test suite.
It also seems to me as if you could reuse the `NetCdf4Test` suite and parameterize on xarray to reduce code duplication. See also https://docs.pytest.org/en/6.2.x/parametrize.html

Reply:
Done.